azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/transport/mqtt/mqtt_transport.py

```diff
@@ -39,13 +68,10 @@ def __init__(self, auth_provider):
         self._mqtt_provider = None
         self.on_transport_connected = None
         self.on_transport_disconnected = None
-        self.on_event_sent = None
-        self._event_queue = queue.LifoQueue()
-        self._event_callback_map = {}
+        self._action_queue = queue.Queue()
```

Comment:
Why the change from LIFO to FIFO? I agree that FIFO seems better, but I wonder why we were even using LIFO to begin with and what the motivation was.

Reply:
Lifo was a mistake.  It should have been fifo all along.

Reply:
sounds reasonable if the goal is to guarantee message order - is this why? 
A small comment about the design choice would protect us from asking again in the future.

---
In reply to: [266153947](https://github.com/Azure/azure-iot-sdk-python-preview/pull/63#discussion_r266153947) [](ancestors = 266153947)